src/lib/strategies/AbstractKeyEventStrategy.js

```diff
@@ -281,20 +286,50 @@ class AbstractKeyEventStrategy {
 
       if (keyMap) {
         Object.keys(keyMap).forEach((actionName) => {
-          keyMapSummary[actionName] = [];
-
-          arrayFrom(keyMap[actionName]).forEach((keySequenceOptions) => {
-            const sequence = function(){
-              if (isObject(keySequenceOptions)) {
-                return keySequenceOptions.sequence;
-              } else {
-                return keySequenceOptions;
-              }
-            }();
+          const keyMapConfig = keyMap[actionName];
 
-            keyMapSummary[actionName].push(sequence);
-          })
-        })
+          keyMapSummary[actionName] = {};
+
+          if (isObject(keyMapConfig)) {
+            if (hasKey(keyMapConfig, 'sequences')) {
+              /**
+               * Support syntax:
+               *  {
+               *    sequences: [ {sequence: 'a+b', action: 'keyup' }],
+               *    name: 'My keymap',
+               *    description: 'Key to press for something special',
+               *    group: 'Vanity'
+               *  }
+               */
+              copyAttributes(
+                keyMapConfig,
+                keyMapSummary[actionName],
+                KEYMAP_ATTRIBUTES
+              );
+
+              keyMapSummary[actionName].sequences =
+                this._createSequenceFromConfig(keyMapConfig.sequences);
+            } else {
+              /**
+               * Support syntax:
+               * {
+               *   sequence: 'a+b', action: 'keyup',
+               *   name: 'My keymap',
+               *   description: 'Key to press for something special',
+               *   group: 'Vanity'
+               * }
+               */
+              copyAttributes(keyMapConfig, keyMapSummary[actionName], KEYMAP_ATTRIBUTES);
```

Comment:
⬜️☂️Explore the possibility of just using `.slice` instead of own utility function. It would mean updating the readme and `index.d.ts`, as all attributes of the object returned by `getApplicationKeyMap()` would always be present.